airbyte_cdk/connector_builder/test_reader/reader.py

```diff
@@ -298,6 +308,17 @@ def _categorise_groups(self, message_groups: MESSAGE_GROUPS) -> GROUPED_MESSAGES
                 case _:
                     raise ValueError(f"Unknown message group type: {type(message_group)}")
 
+        # process deprecation warnings, if present
+        if deprecation_warnings is not None:
+            for deprecation in deprecation_warnings:
+                match deprecation:
+                    case AirbyteLogMessage():
```

Comment:
nit: Why is the typing here different than from line 117? It seems like we wouldn't have to worry about the type not matching here if we could rely on the type.

Reply:
Re-worked. Use the direct type of `LogMessage` as `ConnectorBuilderLogMessage` and append directly. Thanks.